### PR TITLE
Console

### PR DESCRIPTION
**Pull request by tr0ub1eM4k3r**
Closes #{35}

### Changes/Additions:
- Added basic console functionality, with a collapsible state, different tabs and posting strings to the different tabs
-

### CODE REVIEW

**Comment by NybyDK** [comment]
image
Are we doing it wrong or can we not see the bottom ones? Sending the text "Nederst" 5 times but we can't see all of them

https://i.gyazo.com/4e392be4c3c098456aa613483f811ce7.gif
Weird behavior when the window is resized

When selecting the text to copy it, should we get rid of the ">"s, i.e., the arrows should be non-selectable?

When clicking the up arrow, it pushes the rest of the page up, but maybe it should layer on top instead, and not push anything?

**Comment by BaBrixx** [comment]
The bug with not all of the elements showing up in the Console and being able to scroll "below" the bottom of the screen was fixed by removing all of the buttons in the footer so maybe you are calculating the height of the console wrong?

File: src/lib/components/console/Console.svelte [hunk 1]
```diff
@@ -0,0 +1,174 @@
+<script lang="ts">
+	import ConsoleLine from "./ConsoleLine.svelte";
+
+	enum Tabs {
+		Frontend,
+		Backend,
+		All,
+	}
```
Comment (BaBrixx) [inline src/lib/components/console/Console.svelte:8]: Maybe this enum should be in its own file so that it can be used by other components?

**Comment by NybyDK** [comment]
Can we have the frontend/backend buttons show which one is currently used, that is, if I am viewing backend errors, the backend button could have a different color or underline or something

**Review by BaBrixx — CHANGES_REQUESTED** [review]

**Comment by NybyDK** [comment]
In the old GUI, the console can be resized by dragging it at the top to get any size, we should do this too

File: src/lib/components/console/ConsoleLine.svelte [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<script lang="ts">
+	export let componentText: string;
+</script>
+
+<div>
+	<nobr>
+		<p class="unselectable">></p>
+		<p>{componentText}</p>
+	</nobr>
```
Comment (NybyDK) [inline src/lib/components/console/ConsoleLine.svelte:9]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/nobr
> Deprecated (...) it was never standard HTML, so you shouldn't use it. Instead, use the CSS property [white-space](https://developer.mozilla.org/en-US/docs/Web/CSS/white-space)

**Review by NybyDK — CHANGES_REQUESTED** [review]

**Comment by NybyDK** [comment]
For super long errors, it's currently scrolling on the x axis, perhaps it should wrap to the next line but still be under the same ">" arrow, something like

> error here
> another error
> this error is super long and yeehoo wooooooooooooooooo
and it continues on this line
> error :(

**Review by BaBrixx — CHANGES_REQUESTED** [review]
When adding a large amount of elements inside the sidebars, the site reverts to be scrollable and the console is gone.
ezgif-1-41292c2081

**Review by NybyDK — APPROVED** [review]
LOOKS GOOD TO ME

**Review by BaBrixx — APPROVED** [review]
LGTM

**Review by KamyaPA — APPROVED** [review]
CSS classes still in camelCase but approving because of control flow

**Pull request merged**